torch/csrc/jit/python/pybind_utils.h

```diff
@@ -84,6 +84,37 @@ struct VISIBILITY_HIDDEN PythonFunctionGuard {
   py::function func_;
 };
 
+// Handler that runs to raise appropriate python exceptions in future then()
+// callbacks.
+// NB: Need VISIBILITY_HIDDEN for silencing compiler error,
+// 'torch::jit::JitFutureExceptionhandler' declared with greater visibility
+// than the type of its field.
+struct VISIBILITY_HIDDEN JitFutureExceptionHandler {
+  JitFutureExceptionHandler() {
+    DCHECK(PyGILState_Check());
+    const char* kJitFutureExceptionHandlerModule =
+        "torch.jit.exception_handling";
+    const char* kJitFutureExceptionhandlerFunc = "_handle_exception";
```

Comment:
curious, can we try using the python_error defined in [here](https://github.com/pytorch/pytorch/blob/28d6e01511269d894becdf70202dadc63077ba2f/torch/csrc/Exceptions.h#L133) and avoid this manually exception handling?